bvr_ref_compassion/model/invoice.py

```diff
@@ -10,69 +10,63 @@
 ##############################################################################
 import logging
 
-from openerp.osv.orm import Model, fields
+from openerp import models, fields, api, exceptions
 from openerp.tools import mod10r
 
 logger = logging.getLogger(__name__)
 
 
-class account_invoice(Model):
+class account_invoice(models.Model):
 
     ''' Inherit account.invoice in order to change BVR ref field type '''
     _inherit = "account.invoice"
 
-    _columns = {
-        'bvr_reference': fields.char("BVR REF.", size=32,
-                                     track_visibility='onchange')
-    }
+    bvr_reference = fields.Char(
+        "BVR REF.", size=32,
+        track_visibility='onchange')
 
-    def _check_bvr_ref(self, cr, uid, ids, context=None):
-        record = self.browse(cr, uid, ids, context=context)
-        for data in record:
+    @api.constrains('bvr_reference')
+    def _check_bvr_ref(self):
+        for data in self:
             if not data.bvr_reference:
                 return True  # No check if no reference
             clean_ref = data.bvr_reference.replace(' ', '')
             if not clean_ref.isdigit() or len(clean_ref) > 27:
-                return False
+                raise exceptions.ValidationError('Error: BVR ref should only' +
+                                                 'contain number (max. 27) ' +
+                                                 'and spaces.')
             clean_ref = clean_ref.rjust(27, '0')  # Add zeros to the left
             if not clean_ref == mod10r(clean_ref[0:26]):
-                return False
-
+                raise exceptions.ValidationError('Error: BVR ref should only' +

```

Comment:
In this case the error message should rather be : "Invalid BVR ref number. Please check the number."